cleanlab/classification.py

```diff
@@ -428,13 +432,15 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
         x_mask = ~self.label_issues_mask
         x_cleaned = X[x_mask]
         labels_cleaned = labels[x_mask]
+        if sample_weight is not None:
+            sample_weight = sample_weight[x_mask]
         if self.verbose:
             print(f"Pruning {np.sum(self.label_issues_mask)} examples with label issues ...")
             print(f"Remaining clean data has {len(labels_cleaned)} examples.")
 
         # Check if sample_weight in args of clf.fit()
         if (
-            "sample_weight" in inspect.getfullargspec(self.clf.fit).args
+            sample_weight is None
```

Comment:
The main change I would like to see is:  We should not assume `clf.fit()` takes in `sample_weight` as an argument. With your current code, CleanLearning is no longer applicable to users' own bespoke models whose `fit()` function does not take in this argument.

So we should not pass `sample_weight` into `self.clf.fit()` except when it is not None. The easiest way to do that would be to add `sample_weight` to `clf.kwargs` if it is not None.  If we go this route, here are the changes that need to happen:

1) remove `sample_weight` from the `clf.fit()` call.

2) add the following code-block near the start of `CleanLearning.fit()`:

```
if "sample_weight" in clf_kwargs:
    raise ValueError("sample_weight should be provided directly to fit() rather than as clf_kwargs")  # prevent there being multiple ways to pass in sample_weight

if sample_weight is not None:
    if "sample_weight" not in inspect.getfullargspec(self.clf.fit).args:
        raise ValueError("sample_weight must be a supported fit() argument for your model in order to be specified here")
   else:
        clf_kwargs["sample_weight"] = sample_weight
```

Reply:
3) Also cannot delete the check:
```
if "sample_weight" in inspect.getfullargspec(self.clf.fit).args
```
as this will be necessary to determine if cleanlab's built-in sample-weighting can be used or not.

So code for the final clf.fit() on cleaned data should look like:

```
if sample_weight is None:
    if "sample_weight" in inspect.getfullargspec(self.clf.fit).args and "sample_weight" not in self.clf_final_kwargs:
        # use cleanlab's built-in sample-weighting in final clf.fit()...
    else:
        call final clf.fit() without passing sample_weight 
elif sample_weight is not None and "sample_weight" not in self.clf_final_kwargs:
    sample_weight = sample_weight[x_mask]  # I would move this line from above to here
    clf_final_kwargs["sample_weight"] = sample_weight
    # call clf.fit() passing in only clf_final_kwargs, not sample_weight